packages/next/build/utils.ts

```diff
@@ -1037,7 +1052,26 @@ export async function isPageStatic(
       require('../shared/lib/runtime-config').setConfig(runtimeEnvConfig)
       setHttpAgentOptions(httpAgentOptions)
 
-      const mod = await loadComponents(distDir, page, serverless)
+      let mod: any
+
+      if (pageRuntime === SERVER_RUNTIME.edge) {
+        const runtime = await getRuntimeContext({
+          paths: edgeInfo.files.map((file: string) => path.join(distDir, file)),
+          env: edgeInfo.env,
+          edgeFunctionEntry: edgeInfo,
+          name: edgeInfo.name,
+          useCache: true,
+          distDir,
+        })
+        mod =
+          runtime.context._ENTRIES[`middleware_${edgeInfo.name}`].ComponentMod
+
+        mod.Component = mod.default
+        mod.ComponentMod = mod
```

Comment:
maybe use a partial `LoadComponentsReturnType` type like `LoadEdgeComponentsReturnType` to make it safer here?

quite hard to think of why we only need `Component`, `ComponentMod` here. seems mod is already `ComponentMod` then we assign `ComponentMod` to it again?

Reply:
Good point, updated to use the `LoadComponentsReturnType` to ensure we're matching the type correctly although I don't think we need a separate `EdgeComponentsReturnType` since we want to match here.